lib/bolt/transport/ssh/connection.rb

```diff
@@ -228,8 +231,8 @@ def execute(command, sudoable: false, **options)
               channel.on_request("exit-status") do |_, data|
                 result_output.exit_code = data.read_long
               end
-
-              if options[:stdin]
+              # A wrapper is used to direct stdin when elevating privilage or using tty
+              if options[:stdin] && !use_sudo && !options[:wrapper]
                 channel.send_data(options[:stdin])
```

Comment:
I have a local change to do `"option[:stdin]\n"` but it does not seem to make a difference. Seems like it may be safer to do? Either way I will standardize with `handled_sudo`.

Reply:
I think it only makes a difference when using a tty.